### PR TITLE
[Backport v3.7.99-ncs1-branch] [nrf fromlist] soc: nordic: nrf54l15: fix APPROTECT handling

### PR DESCRIPTION
**Pull request by nika-nordic**
Updated sdk-zephyr aligns existing Kconfigs to nRF54L15, while updated TF-M excludes nRF54L15 from existing APPROTECT-related runtime code which is not yet aligned.

This is a backport of https://github.com/nrfconnect/sdk-nrf/pull/18370

### CODE REVIEW

**Review by MarkusLassila — APPROVED** [review]

File: west.yml [hunk 1]
```diff
@@ -153,7 +153,7 @@ manifest:
     - name: trusted-firmware-m
       repo-path: sdk-trusted-firmware-m
       path: modules/tee/tf-m/trusted-firmware-m
-      revision: v2.1.1-ncs1-rc1
+      revision: pull/174/head
```
Comment (rlubos) [inline west.yml:156]: `ac72485fdd643d636808484fcb34dab442dd584d`

File: west.yml [hunk 2]
```diff
@@ -72,7 +72,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: bc59fe3cf65c054b13305aa977ecddddf7cf42e4
+      revision: pull/2211/head
```
Comment (rlubos) [inline west.yml:75]: `0f0e5ad7da0d1e423a1991485e9dfc6fef3757fe`

**Review by rlubos — COMMENTED** [review]
Please update SHAs

**Comment by nika-nordic** [comment]
@rlubos updated SHA as suggested

**Review by rlubos — APPROVED** [review]

**Pull request merged**